classes/navigationMenu/NavigationMenuItem.inc.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/**
+ * @file classes/navigationMenu/NavigationMenuItem.inc.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class NavigationMenuItem
+ * @ingroup navigationMenu
+ * @see NavigationMenuItemDAO
+ *
+ * @brief Basic class describing a NavigationMenuItem.
+ */
+
+/** ID codes and paths for all default navigationMenuItems */
+define('NMI_ID_CURRENT',	0x00000001);
+define('NMI_ID_ARCHIVES',	0x00000002);
+define('NMI_ID_ABOUT',	0x00000003);
+define('NMI_ID_ABOUT_CONTEXT',	0x00000004);
+define('NMI_ID_SUBMISSIONS',	0x00000005);
+define('NMI_ID_EDITORIAL_TEAM',	0x00000006);
+define('NMI_ID_CONTACT',	0x00000007);
+define('NMI_ID_LOGOUT',	0x00000008);
+define('NMI_ID_ANNOUNCEMENTS',	0x00000009);
+
+class NavigationMenuItem extends DataObject {
+	/** @var $navigationMenuItems array The navigationMenuItems underneath this navigationMenuItem */
+	var $navigationMenuItems = array();
+
+	/**
+	 * Constructor
+	 */
+	function __construct() {
+		parent::__construct();
+	}
+
+	//
+	// Get/set methods
+	//
+
+	/**
+	 * Set path for this navigation menu item.
+	 * @param $path string
+	 */
+	function setPath($path) {
+		$this->setData('path', $path);
+	}
+
+	/**
+	 * Get path for this navigation menu item.
+	 * @return string
+	 */
+	function getPath() {
+		return $this->getData('path');
+	}
+
+	/**
+	 * Set page handler name for this navigation menu item.
+	 * @param $page string
+	 */
+	function setPage($page) {
+		$this->setData('page', $page);
+	}
+
+	/**
+	 * Get page handler name for this navigation menu item.
+	 * @return string
+	 */
+	function getPage() {
+		return $this->getData('page');
+	}
+
+	/**
+	 * Set page's op handler name for this navigation menu item.
+	 * @param $op string
+	 */
+	function setOp($op) {
+		$this->setData('op', $op);
+	}
+
+	/**
+	 * Get page's op handler name for this navigation menu item.
+	 * @return string
+	 */
+	function getOp() {
+		return $this->getData('op');
+	}
+
+	/**
+	 * Get is_default for this navigation menu item.
+	 * @return int
+	 */
+	function getDefault() {
+		return $this->getData('is_default');
```

Comment:
I think `symbolic` column would also mean we could get rid of this -- anything with a `symbolic` name would be considered default.